index.html

```diff
@@ -5,16 +5,83 @@
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <meta http-equiv="X-UA-Compatible" content="ie=edge">
-    <link rel="stylesheet" type="text/css" href="css/grid.css">
+    <link rel="stylesheet" type="text/css" href="css/grid.css"> 
+    <link rel="stylesheet" type="text/css" href="css/main.css">
     <!-- Add a link to your custom CSS here -->
 
     <title>Responsive grid project</title>
 </head>
-
 <body>
+<div class="container">
+    <div class="row"> <!--header-->
+        <header>
```

Comment:
Avoid having an element (like your `<header>` here) between the `row` and the `col-...`. The reason for this is that the `row` class has `display: flex` on it, and flexbox only works on immediate, direct children elements.
So here you could get rid of the `div` on line 16, and move your `row` class to be on the `<header>`.